Sources/MapboxMaps/Location/Puck/Puck2D.swift

```diff
@@ -78,54 +100,60 @@ internal final class Puck2D: NSObject, Puck {
     }
 
     private func updateLayer() {
-        guard isActive, let location = locationProducer.latestLocation else {
+        guard isActive, let location = latestLocation else {
             return
         }
-        var layer = LocationIndicatorLayer(id: Self.layerID)
-        layer.location = .constant([
+
+        var newLayerLayoutProperties = [LocationIndicatorLayer.LayoutCodingKeys: Any]()
+        var newLayerPaintProperties = [LocationIndicatorLayer.PaintCodingKeys: Any]()
+
+        newLayerPaintProperties[.location] = [
             location.coordinate.latitude,
             location.coordinate.longitude,
-            location.location.altitude
-        ])
+            location.altitude
+        ]
         switch location.accuracyAuthorization {
         case .fullAccuracy:
-            layer.topImage = .constant(.name(Self.topImageId))
+            let immediateTransition = [
+                StyleTransition.CodingKeys.duration.rawValue: 0,
+                StyleTransition.CodingKeys.delay.rawValue: 0]
+            newLayerLayoutProperties[.topImage] = Self.topImageId
             if configuration.bearingImage != nil {
-                layer.bearingImage = .constant(.name(Self.bearingImageId))
+                newLayerLayoutProperties[.bearingImage] = Self.bearingImageId
             }
-            layer.shadowImage = .constant(.name(Self.shadowImageId))
-            layer.locationTransition = StyleTransition(duration: 0.5, delay: 0)
-            layer.topImageSize = configuration.resolvedScale
-            layer.bearingImageSize = configuration.resolvedScale
-            layer.shadowImageSize = configuration.resolvedScale
-            layer.emphasisCircleRadiusTransition = StyleTransition(duration: 0, delay: 0)
-            layer.bearingTransition = StyleTransition(duration: 0, delay: 0)
+            newLayerLayoutProperties[.shadowImage] = Self.shadowImageId
+            newLayerPaintProperties[.locationTransition] = immediateTransition
```

Comment:
I think gl-native supports flattened layer properties, so no need to separate the layout and paint layer properties separately.

Reply:
It's separated here to make the implementation cleaner. The key types are paint and layout specific: https://github.com/mapbox/mapbox-maps-ios/pull/1090/files#diff-49c72aa33067d00c9bc906b7e1e0d4e6d56bd77db95450efcf80115bc157307fR107-R108

Later, the keys are mapped into strings. We could use strings at this step, but it would either mean making the code much more verbose or using raw strings in the implementation.